packages/ngtools/webpack/src/ivy/plugin.ts

```diff
@@ -304,6 +302,26 @@ export class AngularWebpackPlugin {
     });
   }
 
+  private markResourceUsed(
+    normalizedResourcePath: string,
+    currentUnused: Set<string>,
+    visitedResources: Set<string>,
+  ): void {
+    if (visitedResources.has(normalizedResourcePath)) {
```

Comment:
Why do we need to keep track of `visitedResources` separately? We can use `currentUnused` as our single source of truth.

```ts
if (!currentUnused.has(normalizedResourcePath)) {
    return;
}

currentUnused.delete(normalizedResourcePath);

const dependencies = this.fileDependencies.get(normalizedResourcePath);
if (!dependencies) {
    return;
}

for (const dependency of dependencies) {
    this.markResourceUsed(normalizePath(dependency), currentUnused);
}
```